### DISCUSSION TITLE
Define rgb band map

### DISCUSSION THREAD

**Discussion by charalamm** [developmentseed/titiler#672] [Q&A] [answered]
Hello,

I have a raster with three bands which are the red, green, blue band in that order. When I try to view the raster in titiler its gray. Is there a way to instruct titiler to use the bands for coloring the raster?

I have tried the `rescale` and `bidx` (not sure they are related in any way) but they did not work.

**Comment by vincentsarago** [comment]
@charalamm It's pretty hard for us to help without having more info about the raster so I'll make guess... 

if the tiles appear grey in the application it might be because your raster is not in `Uint8` datatype, which means that titiler will have to automatically rescale the value from the original dataset to `Uint8` to be able to create proper PNG/JPEG tiles. To do this, titiler will try to see if there are min/max value within the file metadata to `rescale` using those values, if not found it will take the min/max of the data type.

Note: if your data is in Uint16, titiler might not try to rescale the data because Uint16 is supported by PNG (but not supported by all client application). 

Anyway, when your data is not in Uint8 datatype you better use `rescale={min},{max}` query-parameter option, but you said it didn't work... which then gives me two options:
- you min/max options are wrong
- there is a possible nodata issue, but I can't tell without more information 🤷

**Reply by charalamm** [reply]
@vincentsarago Thanks for your quick response once again! Sorry for not providing enough data. [This](https://savmystorage.blob.core.windows.net/my-sav-titiler/quicklooks/region_0001/2023-01-02/0001_2023-01-02.tif) is the raster and as you foreseen it is int16 instead int8. The maximum value among all bands is 5492, so I tried with [https://titiler.xyz/cog/preview.png?url=https://savmystorage.blob.core.windows.net/my-sav-titiler/quicklooks/region_0001/2023-01-02/0001_2023-01-02.tif?rescale=0,5492](https://titiler.xyz/cog/preview.png?url=https://savmystorage.blob.core.windows.net/my-sav-titiler/quicklooks/region_0001/2023-01-02/0001_2023-01-02.tif?rescale=0,5492) btu it did not work.

Bellow is an image from the same raster in QGIS:
![image](https://github.com/developmentseed/titiler/assets/64330098/3369803f-98a1-4149-b990-de19716d12b1)

**Reply by vincentsarago** [reply]
thanks @charalamm 

there is definitely something weird, I'll investigate 🙏

**Reply by vincentsarago** [reply, marked as answer]
correct url is 
```
https://titiler.xyz/cog/preview.png?url=https://savmystorage.blob.core.windows.net/my-sav-titiler/quicklooks/region_0001/2023-01-02/0001_2023-01-02.tif&rescale=0,5492
```

`&rescale=0,5492` instead of `?rescale=0,5492`

**Reply by charalamm** [reply]
My bad 🤦. Thank you for your time!
